client/asset/dcr/coin_selection.go

```diff
@@ -165,37 +185,78 @@ searchLoop:
 //
 // If the provided UTXO set has less combined value than the requested amount a
 // nil slice is returned.
-func leastOverFund(amt uint64, utxos []*compositeUTXO) []*compositeUTXO {
-	if amt == 0 || sumUTXOs(utxos) < amt {
-		return nil
+func leastOverFund(enough func(sum uint64, size uint32, unspent *compositeUTXO) (bool, uint64), utxos []*compositeUTXO) []*compositeUTXO {
+	return leastOverFundWithLimit(enough, math.MaxUint64, utxos)
+}
+
+// enoughWithoutAdditional is used to utilize an "enough" function with a set
+// of UTXOs when we are not looking to add another UTXO to the set, just
+// check if the current set if UTXOs is enough.
+func enoughWithoutAdditional(enough func(sum uint64, size uint32, unspent *compositeUTXO) (bool, uint64), utxos []*compositeUTXO) bool {
+	if len(utxos) == 0 {
+		return false
+	}
+
+	if len(utxos) == 1 {
+		e, _ := enough(0, 0, utxos[0])
+		return e
+	}
+
+	valueWithoutLast := sumUTXOs(utxos[:len(utxos)-1])
+	sizeWithoutLast := sumUTXOSize(utxos[:len(utxos)-1])
```

Comment:
Why does this exclude the last utxo?

Reply:
Because the `enough` functions in DCR check if adding a new UTXO to the existing set will make it enough, but here I'm just trying to check if the entire set is enough or not. So I just take the entire set without the last element, and use the last element as the new UTXO in the enough function.

Reply:
Oh, I was over thinking this. I get it now.